daemon/mgr/container.go

```diff
@@ -450,18 +450,48 @@ func (mgr *ContainerManager) Start(ctx context.Context, id, detachKeys string) (
 }
 
 func (mgr *ContainerManager) start(ctx context.Context, c *Container, detachKeys string) error {
+	var err error
+
 	c.Lock()
 	if c.Config == nil || c.State == nil {
 		c.Unlock()
 		return errors.Wrapf(errtypes.ErrNotfound, "container %s", c.ID)
 	}
 	c.DetachKeys = detachKeys
 
+	// attach volume
+	for _, mp := range c.Mounts {
+		if mp.Name == "" {
+			continue
+		}
+
+		attachedVolumes := map[string]struct{}{}
```

Comment:
is it necessary to add a defer in a loop

Reply:
oh, it should defer outside the loop